### PR TITLE
Change name hnsw and add HNSW indexing config

### PR DESCRIPTION
**Pull request by MXueguang**
now we can config hnsw parameters
```
./target/appassembler/bin/IndexDenseVectors -collection JsonDenseVectorCollection \
 -generator LuceneDenseVectorDocumentGenerator -threads 1 -input <> \
 -index <> -M 32 -efC 100
```

### CODE REVIEW

**Comment by MXueguang** [comment]
also solve part of https://github.com/castorini/anserini/issues/2018

File: src/main/java/io/anserini/index/IndexDenseVectorsArgs.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
 import org.kohsuke.args4j.spi.StringArrayOptionHandler;
 
 
-public class IndexVectorArgs {
+public class IndexDenseVectorsArgs {
```
Comment (lintool) [inline src/main/java/io/anserini/index/IndexDenseVectorsArgs.java:23]: I thought we wanted to make this an inner class?

File: src/main/java/io/anserini/search/SearchDenseVectors.java [hunk 2]
```diff
@@ -316,12 +316,12 @@ public static void main(String[] args) throws Exception {
...
 
     final long start = System.nanoTime();
-    SearchVectorCollection searcher;
+    SearchDenseVectors searcher;
 
     // We're at top-level already inside a main; makes no sense to propagate exceptions further, so reformat the
     // exception messages and display on console.
     try {
-      searcher = new SearchVectorCollection(searchArgs);
+      searcher = new SearchDenseVectors(searchArgs);
```
Comment (lintool) [inline src/main/java/io/anserini/search/SearchDenseVectors.java:324]: Same of the args here... did we want to make them inner classes?

File: src/main/java/io/anserini/index/IndexDenseVectors.java [hunk 3]
```diff
@@ -364,19 +372,19 @@ public Counters run() throws IOException {
   }
 
   public static void main(String[] args) throws Exception {
-    IndexVectorArgs indexCollectionArgs = new IndexVectorArgs();
+    IndexDenseVectorsArgs indexCollectionArgs = new IndexDenseVectorsArgs();
```
Comment (MXueguang) [inline src/main/java/io/anserini/index/IndexDenseVectors.java:375]: a java question

I'm a bit unsure how to properly initialize the Args here if we make it subclass
@lintool 
I don't think I can do
```
IndexDenseVectors.Args args = new IndexDenseVectors.Args()
```

Comment (MXueguang) [reply src/main/java/io/anserini/index/IndexDenseVectors.java:375]: we need to parse the args first and then initialize the index class as IndexDenseVectors(args)

**Review by lintool — APPROVED** [review]

**Pull request merged**